Fails to build with GCC 15 (defaults to -std=gnu23)

Upcoming GCC 15 defaults to `-std=gnu23`. p11-kit-0.25.5 fails to build like:
```
FAILED: p11-kit/p11-kit.p/export-object.c.o
x86_64-pc-linux-gnu-gcc -m32 -mfpmath=sse -Ip11-kit/p11-kit.p -Ip11-kit -I../p11-kit-0.25.5/p11-kit -I. -I../p11-kit-0.25.5 -Icommon -I../p11-kit-0.25.5/common -I/usr/lib/libffi/include
-flto=auto -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -D_GNU_SOURCE -DP11_KIT_FUTURE_UNSTABLE_API -O3 -march=native -mtls-dialect=gnu2 -fno-semantic-interposition -pipe -fcf-protection=none -fdiagnostics-color=always -fdiagnostics-urls=never -frecord-gcc-switches -fdiagnostics-details -Wa,-O2 -Wa,-mtune=znver2 -Warray-bounds -Wstrict-aliasing -Wfree-nonheap-object -Wuninitialized -Wmaybe-uninitialized -Wreturn-type -Werror=lto-type-mismatch -Werror=strict-aliasing -Werror=odr -Warray-bounds -fdiagnostics-details -Wstringop-overread -Wstringop-overflow -Wstringop-truncation -Wbuiltin-declaration-mismatch -ggdb3 -Wformat -Werror=format-security '-DBINDIR="/usr/bin"' '-DPRIVATEDIR="/usr/libexec/p11-kit"' '-DSYSCONFDIR="/etc"' '-DP11_SYSTEM_CONFIG_FILE="/etc/pkcs11/pkcs11.conf"' '-DP11_SYSTEM_CONFIG_MODULES="/etc/pkcs11/modules"' '-DP11_PACKAGE_CONFIG_MODULES="/usr/share/p11-kit/modules"' '-DP11_USER_CONFIG_FILE="~/.config/pkcs11/pkcs11.conf"' '-DP11_USER_CONFIG_MODULES="~/.config/pkcs11/modules"' '-DP11_MODULE_PATH="/usr/lib/pkcs11"' -MD -MQ p11-kit/p11-kit.p/export-object.c.o -MF p11-kit/p11-kit.p/export-object.c.o.d -o p11-kit/p11-kit.p/export-object.c.o -c ../p11-kit-0.25.5/p11-kit/export-object.c
In file included from ../p11-kit-0.25.5/p11-kit/export-object.c:43:
../p11-kit-0.25.5/p11-kit/export-object.c: In function ‘export_pubkey_ec’:
../p11-kit-0.25.5/p11-kit/export-object.c:237:36: error: incompatible types when returning type ‘_Bool’ but ‘unsigned char *’ was expected
  237 |         return_val_if_fail (attrs, false);
      |                                    ^~~~~
../p11-kit-0.25.5/common/debug.h:76:21: note: in definition of macro ‘return_val_if_fail’
   76 |              return v; \
      |                     ^
```

--- a/p11-kit/export-object.c
+++ b/p11-kit/export-object.c
@@ -234,7 +234,7 @@ export_pubkey_ec (const ExportData *data,
 	unsigned char *der = NULL;
 
 	attrs = p11_attrs_buildn (NULL, template, p11_attrs_count (template));
-	return_val_if_fail (attrs, false);
+	return_val_if_fail (attrs, NULL);
 
 	if (p11_kit_iter_load_attributes (data->iter,
 					  attrs,

--- a/trust/x509.c
+++ b/trust/x509.c
@@ -122,7 +122,7 @@ p11_x509_parse_subject_key_identifier  (p11_dict *asn1_defs,
 	unsigned char *keyid;
 	asn1_node ext;
 
-	return_val_if_fail (keyid_len != NULL, false);
+	return_val_if_fail (keyid_len != NULL, NULL);
 
 	ext = p11_asn1_decode (asn1_defs, "PKIX1.SubjectKeyIdentifier", ext_der, ext_len, NULL);
 	if (ext == NULL)
@@ -259,7 +259,7 @@ p11_x509_parse_directory_string (const unsigned char *input,
 	return_val_if_fail (ret == ASN1_SUCCESS, NULL);
 
 	octet_len = asn1_get_length_der (input + tag_len, input_len - tag_len, &len_len);
-	return_val_if_fail (octet_len >= 0, false);
+	return_val_if_fail (octet_len >= 0, NULL);
 	return_val_if_fail (tag_len + len_len + octet_len == input_len, NULL);
 
 	octets = input + tag_len + len_len;

